cow_instrument/FlatTree.h

```diff
@@ -0,0 +1,75 @@
+#ifndef FLATTREE_H
+#define FLATTREE_H
+
+#include <memory>
+#include <vector>
+#include <map>
+#include <Eigen/Core>
+#include "IdType.h"
+#include "ComponentInfo.h"
+
+class Detector;
+class PathComponent;
+class PointSource;
+class PointSample;
+
+/*
+ Flattened representation of an Instrument Tree.
+ */
+class FlatTree {
```

Comment:
Apart from a few exceptions, `FlatTree` mostly forwards to `ComponentInfo`. Not having looked at everything in detail it is not clear to me why we need two separate classes here. What is the motiviation?